From 01c377f7da9b3c4c36cc9f764e0d133aab619e64 Mon Sep 17 00:00:00 2001
From: Petr Oros <poros@redhat.com>
Date: Thu, 5 Sep 2019 10:01:02 -0400
Subject: [PATCH 06/96] [netdrv] bnxt: call devlink_port_type_eth_set() before
 port register

Message-id: <cb34fbcb1af1e95d97609936c5d091d1043eca08.1567595345.git.poros@redhat.com>
Patchwork-id: 270943
O-Subject: [RHEL8.2 net PATCH 06/15] bnxt: call devlink_port_type_eth_set() before port register
Bugzilla: 1737133
RH-Acked-by: Jonathan Toppins <jtoppins@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Bugzilla: http://bugzilla.redhat.com/show_bug.cgi?id=1737133

Upstream commit(s):
commit c3f10cbcaa3d5e1980733c3ccd0261df426412d2
Author: Jiri Pirko <jiri@mellanox.com>
Date:   Sun Mar 24 11:14:28 2019 +0100

    bnxt: call devlink_port_type_eth_set() before port register

    Call devlink_port_type_eth_set() before devlink_port_register(). Bnxt
    instances won't change type during lifetime. This avoids one extra
    userspace devlink notification.

    Signed-off-by: Jiri Pirko <jiri@mellanox.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Petr Oros <poros@redhat.com>
Signed-off-by: Timothy Redaelli <tredaelli@redhat.com>
Signed-off-by: Bruno Meneguele <bmeneg@redhat.com>
---
 drivers/net/ethernet/broadcom/bnxt/bnxt_devlink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: src/drivers/net/ethernet/broadcom/bnxt/bnxt_devlink.c
===================================================================
--- src.orig/drivers/net/ethernet/broadcom/bnxt/bnxt_devlink.c	2020-02-06 16:23:11.535551475 +0100
+++ src/drivers/net/ethernet/broadcom/bnxt/bnxt_devlink.c	2020-02-06 16:23:11.657550355 +0100
@@ -229,6 +229,7 @@
 		goto err_dl_unreg;
 	}
 
+	devlink_port_type_eth_set(&bp->dl_port, bp->dev);
 	devlink_port_attrs_set(&bp->dl_port, DEVLINK_PORT_FLAVOUR_PHYSICAL,
 			       bp->pf.port_id, false, 0);
 	rc = devlink_port_register(dl, &bp->dl_port, bp->pf.port_id);
@@ -236,7 +237,6 @@
 		netdev_err(bp->dev, "devlink_port_register failed");
 		goto err_dl_param_unreg;
 	}
-	devlink_port_type_eth_set(&bp->dl_port, bp->dev);
 
 	rc = devlink_port_params_register(&bp->dl_port, bnxt_dl_port_params,
 					  ARRAY_SIZE(bnxt_dl_port_params));
